### PR TITLE
Added ncm commands

### PR DESCRIPTION
**Pull request by Adubbz**
This is based on https://github.com/switchbrew/libnx/pull/104/ with several additions of my own.

### CODE REVIEW

File: nx/include/switch/services/ncm.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * @brief Content Manager (ncm) service IPC wrapper.
+ * @copyright libnx Authors
+ */
+#pragma once
+#include "../types.h"
+#include "../services/fs.h"
+#include "../services/sm.h"
+
+typedef enum  {
+    NCMContentType_CNMT = 0,
```
Comment (fincs) [inline nx/include/switch/services/ncm.h]: `NcmContentType_*`

File: nx/include/switch/services/ncm.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/**
+ * @file ncm.h
+ * @brief Content Manager (ncm) service IPC wrapper.
+ * @copyright libnx Authors
```
Comment (fincs) [inline nx/include/switch/services/ncm.h]: Should add ` @author zhuowei & Adubbz` before this line.

**Comment by fincs** [comment]
Thank you for continuing development of the previous PR, addressing the comments we made.

File: nx/include/switch/services/ncm.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+typedef struct {
+    NcmMetaRecord metaRecord;
+    u64           baseTitleId;
+} NcmApplicationContentMetaKey;
+
+typedef struct {
+    u8 c[0x10];
+} NcmRightsId;
+
+Result ncmInitialize();
```
Comment (fincs) [inline nx/include/switch/services/ncm.h]: `(void)` here and in the line below

File: nx/include/switch/services/ncm.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+Result ncmContentMetaDatabaseGet(NcmContentMetaDatabase* db, const NcmMetaRecord* record, u64 contentRecordSize, NcmContentRecord* contentRecordsOut, u64* sizeRead);
+Result ncmContentMetaDatabaseRemove(NcmContentMetaDatabase* db, const NcmMetaRecord *record);
+Result ncmContentMetaDatabaseGetContentIdByType(NcmContentMetaDatabase* db, NcmContentType contentType, const NcmMetaRecord* record, NcmNcaId* out);
+Result ncmContentMetaDatabaseListContentInfo(NcmContentMetaDatabase* db, const NcmMetaRecord* record, u32 index, NcmContentRecord* contentRecordsOut, size_t contentRecordsBufSize, u32* numEntriesRead);
+Result ncmContentMetaDatabaseGetLatestContentMetaKey(NcmContentMetaDatabase* db, u64 titleId, NcmMetaRecord* out);
+Result ncmContentMetaDatabaseListApplication(NcmContentMetaDatabase* db, u8 filter, NcmApplicationContentMetaKey* outBuf, size_t outBufSize, u32* numEntriesWritten, u32* numEntriesTotal);
+Result ncmContentMetaDatabaseHas(NcmContentMetaDatabase* db, const NcmMetaRecord* record, bool* out);
+Result ncmContentMetaDatabaseCommit(NcmContentMetaDatabase* db);
```
Comment (fincs) [inline nx/include/switch/services/ncm.h]: Missing newline at line end

File: nx/source/services/ncm.c [hunk 5]
```diff
@@ -0,0 +1,607 @@
...
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+    }
+    
+    return rc;
+}
```
Comment (fincs) [inline nx/source/services/ncm.c]: Missing newline at line end

File: nx/source/services/ncm.c [hunk 6]
```diff
@@ -0,0 +1,607 @@
...
+            u64 magic;
+            u64 result;
+            NcmRightsId rights_id;
+            u32 key_generation;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+
+        if (R_SUCCEEDED(rc)) {
+            memcpy(rightsIdOut, &resp->rights_id, sizeof(NcmRightsId));
```
Comment (yellows8) [inline nx/source/services/ncm.c]: Add if-block like the below line.

File: nx/source/services/ncm.c [hunk 7]
```diff
@@ -0,0 +1,607 @@
+#include "services/ncm.h"
+#include <string.h>
```
Comment (yellows8) [inline nx/source/services/ncm.c]: Move to the top.

File: nx/source/services/ncm.c [hunk 8]
```diff
@@ -0,0 +1,607 @@
...
+
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+
+        if (R_SUCCEEDED(rc)) {
+            strncpy(out, out_path, FS_MAX_PATH);
```
Comment (yellows8) [inline nx/source/services/ncm.c]: Change size to ncmContentStorageGetPath new param out_size. Clamp the size to FS_MAX_PATH-1.

**Pull request merged**